pydantic_ai_slim/pydantic_ai/exceptions.py

```diff
@@ -168,3 +168,12 @@ class ToolRetryError(Exception):
     def __init__(self, tool_retry: RetryPromptPart):
         self.tool_retry = tool_retry
         super().__init__()
+
+def iter_exc(exc):
```

Comment:
Since we currently only using this in one place, I'd prefer for it to be defined as a private method in `mcp.py`. The name could be a bit clearer, and it'll need an argument type hint as well.

Reply:
Actually, since we only use it one place where we know we have an `ExceptionGroup`, could we inline it?